Sprint_2

lexlobov opened the pull request.

lexlobov committed:
Переделал весь проект на новый репозиторий

egorvas commented on src/main/java/com/example/AlexTheLion.java:
```diff
@@ -0,0 +1,20 @@
+    public AlexTheLion() throws Exception{
+        super("Самец");
+    }
+    private final String sex1 = "Самец";
```
Давай без цифр в названии констант и переменных, тут это совсем не нужно. Да и вообще я не понимаю зачем тебе эта константа.

egorvas commented on src/main/java/com/example/Lion.java:
```diff
@@ -2,7 +2,7 @@
 import java.util.List;
 
-public class Lion {
+public class Lion extends Feline {
```
Ты явно не верно понял пункт про изоляцию feline. Нужно было не полностью избавиться от feline, а вынести его в конструктор класса Lion. Т.е. под изоляцией понимается то, что объект feline создается отдельно и уже готовый объект передается в конструкторе Lion.

egorvas commented on src/test/java/AnimalTest.java:
```diff
@@ -0,0 +1,39 @@
+    @Test
+    public void getFoodThrowsException() throws Exception{
+        Animal animal = new Animal();
+        try {
```
Если ты хочешь проверять эксепшен через try/catch, то тебе обязательно нужно добавить проверку на то что эксепшен произошел. Потому что сейчас если эксепшен вдруг не произойдет, то твой тест пройдет без ассертов.

egorvas commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,34 @@
+    public void getFoodReturnsPredatorFood(){
+        List<String> expectedResult = List.of("Животные", "Птицы", "Рыба");
+        Cat cat = new Cat(feline);
+        try {
```
Здесь не нужен try/catch, твой тест и так упадет если что-то попойдет не так

egorvas commented on src/test/java/FelineTest.java:
```diff
@@ -0,0 +1,43 @@
+    public void eatMeatReturnsPredatorList(){
+        Feline feline = new Feline();
+        List<String> expectedResult = List.of("Животные", "Птицы", "Рыба");
+        try {
```
Тут тоже не нужен try/catch

egorvas commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,18 @@
+    @Test
+    public void lionThrowsException(){
+        try {
+            Lion lion = new Lion(wrongSex);
```
Здесь тоже нужно проверять что твой catch произошел. Иначе твой тест никогда не упадет.

lexlobov committed:
Убрал try catch там где он не был нужен

lexlobov committed:
Убрал try catch там где он не был нужен

lexlobov committed:
- убрал ненужную переменную
- переделал класс Lion
- переделал класс AlexTheLion
- Добавил тесты для класса Lion
- Переработал проверку исключений
- Убрал ненужные try catch

lexlobov committed:
- убрал ненужную переменную
- переделал класс Lion
- переделал класс AlexTheLion
- Добавил тесты для класса Lion
- Переработал проверку исключений
- Убрал ненужные try catch

egorvas commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,36 @@
+public class LionTest {
+    private final String wrongSex = "1234";
+    private final String maleSex = "Самец";
+    @Mock
```
Не забывай о пустых строках между кусками кода, читать вот эту конструкцию как у тебя очень неудобно

lexlobov committed:
хз
